runtime/src/installed_scheduler_pool.rs

```diff
@@ -27,6 +100,9 @@ use {mockall::automock, qualifier_attr::qualifiers};
     allow(unused_attributes, clippy::needless_lifetimes)
 )]
 pub trait InstalledScheduler: Send + Sync + Debug + 'static {
+    fn id(&self) -> SchedulerId;
+    fn pool(&self) -> InstalledSchedulerPoolArc;
```

Comment:
Do we ever need/use the pool for anything other than a call to `return_to_pool`?

If that's the case, why not just have a direct function on this trait that consumes the appropriate wrapped type (`Box<Self>`, I think)?

Reply:
wow, super nice suggestion. this resulted in sizable code reorg: c15aac3c4b9d84e32cdfe8afe72ee39972d73779 2cac9721e48ff09a060a468631ccd5d2d09eae42 d4aee4317fe9384c55776c54135773f0c16d0a7e